library/core/src/slice/mod.rs

```diff
@@ -882,6 +882,36 @@ impl<T> [T] {
         ChunksExactMut::new(self, chunk_size)
     }
 
+    /// Splits the slice into a slice of `N`-element arrays,
+    /// starting at the beginning of the slice,
+    /// and a remainder slice with length strictly less than `N`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `N` is 0. This check will most probably get changed to a compile time
+    /// error before this method gets stabilized.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_as_chunks)]
+    /// let slice = ['l', 'o', 'r', 'e', 'm'];
+    /// let (chunks, remainder) = slice.as_chunks();
+    /// assert_eq!(chunks, &[['l', 'o'], ['r', 'e']]);
+    /// assert_eq!(remainder, &['m']);
+    /// ```
+    #[unstable(feature = "slice_as_chunks", issue = "74985")]
+    #[inline]
+    pub fn as_chunks<const N: usize>(&self) -> (&[[T; N]], &[T]) {
+        assert_ne!(N, 0);
+        let len = self.len() / N;
+        let (fst, snd) = self.split_at(len * N);
```

Comment:
Not a fan of the `fst` and `snd` style of abbreviation. I think it's rarely used in this codebase and is generally discouraged. Also, `remainder` might be a better name for the second part. Applies to the `_mut` version as well.

Reply:
@LukasKalbertodt I asked what is `fst` and `snd` in the past and it was encouraged in the pull request, I forgot which pull request was it.

Reply:
Okay, interesting. I never saw these specific abbreviations in std's codebase. (Doesn't mean they don't exist, of course)

Reply:
Just for reference https://github.com/rust-lang/rust/pull/75936#discussion_r478542156 and https://github.com/rust-lang/rust/pull/75936#discussion_r477363761, I personally got confused the first time I saw it.

Reply:
These names are just moved over from the previous code.  I'm not tied to them; new commit to update them incoming.